bootstrap/Makefile.bootstrap

```diff
@@ -2,19 +2,64 @@ export SHELL = /bin/bash
 export TARDIGRADE_CI_ORG ?= plus3it
 export TARDIGRADE_CI_PROJECT ?= tardigrade-ci
 export TARDIGRADE_CI_BRANCH ?= $(or $(shell grep 'FROM $(TARDIGRADE_CI_ORG)/$(TARDIGRADE_CI_PROJECT)' Dockerfile | grep -oE '[0-9]+\.[0-9]+\.[0-9]+' 2> /dev/null),master)
-export TARDIGRADE_CI_PATH ?= $(shell until [ -d "$(TARDIGRADE_CI_PROJECT)" ] || [ "`pwd`" == '/' ]; do cd ..; done; pwd)/$(TARDIGRADE_CI_PROJECT)
 
--include $(TARDIGRADE_CI_PATH)/Makefile
+TARDIGRADE_CI_PATH ?= $(shell until [ -d "$(TARDIGRADE_CI_PROJECT)" ] || [ "`pwd`" == '/' ]; do cd ..; done; pwd)/$(TARDIGRADE_CI_PROJECT)
+TARDIGRADE_CI_PATH_LOCAL := $(PWD)/$(TARDIGRADE_CI_PROJECT)
+TARDIGRADE_CI_BOOTSTRAP := $(PWD)/.$(TARDIGRADE_CI_PROJECT)
+TARGIDRADE_CI_CLONE_URL ?= https://github.com/$(TARDIGRADE_CI_ORG)/$(TARDIGRADE_CI_PROJECT).git
+
+# Resolves to TARDIGRADE_CI_PATH_LOCAL when TARDIGRADE_CI_PATH does not exist
+export TARDIGRADE_CI_PATH := $(or $(wildcard $(TARDIGRADE_CI_PATH)),$(TARDIGRADE_CI_PATH_LOCAL))
+TARDIGRADE_CI_MAKEFILE := $(TARDIGRADE_CI_PATH)/Makefile
+
+# Macro to clone TARDIGRADE_CI_PROJECT
+define $(TARDIGRADE_CI_PROJECT)/clone
+rm -rf "$(TARDIGRADE_CI_PATH_LOCAL)" ;\
+echo "Cloning $(TARGIDRADE_CI_CLONE_URL)?ref=$(TARDIGRADE_CI_BRANCH)..." >&2 ;\
+git clone -c advice.detachedHead=false --depth=1 -b "$(TARDIGRADE_CI_BRANCH)" "$(TARGIDRADE_CI_CLONE_URL)"
+endef
+
+# Macro to auto-init the TARDIGRADE_CI_PROJECT with the `include` directive
+# Tests if TARDIGRADE_CI_PROJECT does not yet exist, or if it does exist but the
+# checkout does not match TARDIGRADE_CI_BRANCH
+# editorconfig-checker-disable
+define $(TARDIGRADE_CI_PROJECT)/init
+if [[ \
+  "$(TARDIGRADE_CI_PATH)" != "$(TARDIGRADE_CI_PATH_LOCAL)" && \
+  -f "$(TARDIGRADE_CI_MAKEFILE)" \
+]]; then \
+  : ;\
+elif [[ \
+  "$(TARDIGRADE_CI_PATH)" == "$(TARDIGRADE_CI_PATH_LOCAL)" && \
+  -f "$(TARDIGRADE_CI_MAKEFILE)" && \
+  $$(git -C $(TARDIGRADE_CI_PATH_LOCAL) ls-remote "$(TARGIDRADE_CI_CLONE_URL)" "$(TARDIGRADE_CI_BRANCH)" | cut -f1) == $$(git -C $(TARDIGRADE_CI_PATH_LOCAL) rev-parse HEAD) \
+]]; then \
+  : ;\
+else \
+  $($(TARDIGRADE_CI_PROJECT)/clone) ;\
+fi
+endef
```

Comment:
endef -> endif; in more than one place.

Reply:
I don't know why they chose this marker, it is not intuitive, and easily looks like a typo, but `endef` is the closing syntax for the Make `define` directive. 

* https://www.gnu.org/software/make/manual/html_node/Multi_002dLine.html